Neos.Neos/Classes/Service/LinkingService.php

```diff
@@ -115,24 +118,24 @@ class LinkingService
     protected $siteRepository;
 
     /**
-     * @param string|Uri $uri
+     * @param string|UriInterface $uri
      * @return boolean
      */
     public function hasSupportedScheme($uri): bool
     {
-        if ($uri instanceof Uri) {
+        if ($uri instanceof UriInterface) {
             $uri = (string)$uri;
         }
         return preg_match(self::PATTERN_SUPPORTED_URIS, $uri) === 1;
```

Comment:
```suggestion
        return preg_match(self::PATTERN_SUPPORTED_URIS, (string)$uri) === 1;
```
This should do the trick. Of course the lines above can then be removed even.